src/Enzo/enzo_EnzoInitialShockTube.cpp

```diff
@@ -0,0 +1,329 @@
+// See LICENSE_CELLO file for license and copyright information
+
+/// @file     enzo_EnzoInitialShockTube.cpp
+/// @author   Matthew Abruzzo (matthewabruzzo@gmail.com)
+/// @date     2019-04-18
+/// @brief    Implementation of EnzoInitialShockTube for initializing the RJ2a
+///           shocktube problem mentioned by Stone et al 2008 and detailed in
+///           Ryu & Jones (1995). This could probably be modified to include
+///           additional shock tubes
+
+// Note that there are a number of typos in various papers describing the
+// initial conditions for the Ryu & Jones (95) 2a shock tube However, I think I
+// have converged on the correct initial conditions from agreement between Dai
+// & Woodward (93) (which lists the expected soln) and Stone et al. (08)
+// Typos:
+//   Ryu & Jones (95)
+//     - states that the x-component of the magnetic field is of 2(4pi)^.5. It
+//       should actually be 2/(4pi)^0.5
+//     - They state that the left and right energies have values of 0.95 and 1.
+//       Per Dai & Woodward (93) and Stone et al. (08) the left and right
+//       pressures have values of 0.95 and 1.0.
+//   Gardiner & Stone (08)
+//     - states that both the left and right states have pressures of 0.95. Dai
+//       & Woodward (93) and Stone et al. (08) indicate that the left state has 
+//       a pressure of 0.95 and the right state has a pressure of 1.
+
+#include "cello.hpp"
+#include "enzo.hpp"
+#include <math.h> // ceil
+
+static std::vector<std::string> shock_tube_setups{"rj2a","sod"};
+
+
+
+// right states for various shock tubes
+static std::map<std::string, std::map<std::string, enzo_float>> shock_tube_l{
+  {"rj2a", {// Ryu & Jones (95) from 2a
```

Comment:
I totally agree.

I think the reason I didn't do this was because I didn't want to add a ton of parameters to the configuration file.

But as I think about it now, there is an obvious way to do this while just adding 2 parameters (one for the left states and one for the right states).